accepted/2025/Multiple-Equivalent-Framework-Support-TFM-As-Aliases.md

```diff
@@ -0,0 +1,463 @@
+The current assets file version is 3, and this feature will increment the version to 4.
+
+```diff
+-"version": 3
++"version": 4
+```
+
+By default for non-SDK style projects, we will always use version 3.
+Non-SDK style project only have a single target, so those project types won't see a change.
+
+For simplicity of implementation, whenever SDKAnalysisLevel is set to 10.0.200 or more, NuGet will use the V4 assets file.
+Whenever the SDKAnalysisLevel is set to a lower version, then NuGet will use the V3 assets file.
+Scenarios where the .NET SDK is newer of the Visual Studio version are not a concern.
+The same considerations extend to NuGet.exe restore, which will also be dependent on the SDKAnalysisLevel property.
+
+#### Target Framework Pivots
+
+There are multiple places in the assets file where the "NuGet Framework" is used as a JSON object property key.
+There are 3 top level sections where changes will be needed. The `targets`, `projectFileDependencyGroups` and `project` sections.
+
+Take a project using `<TargetFramework>production</TargetFramework>`, `<TargetFrameworkMoniker>.NETCoreApp,Version=v8.0</TargetFrameworkMoniker>`, and `<RuntimeIdentifiers>linux-x64</RuntimeIdentifiers>`.
+
+Note that the effective framework itself will not be part of the serialized part, but it'll cleaned up by the reader by looking up the target framework based off of the "project" section of the assets file.
+
+1. `$/targets`
+
+    ```diff
+      "targets": {
+    -    "net8.0": {},
+    -    "net8.0/linux-x64": {},
+    +    "production": {},
+    +    "production/linux-x64": {},
+      },
+    ```
+
+For the projectFileDependencyGroup, the type already uses a string as the pivot, which simplifies things quite a bit, <https://github.com/NuGet/NuGet.Client/blob/f764caebef5937cff2802ba65208e8f43afc6a45/src/NuGet.Core/NuGet.ProjectModel/ProjectFileDependencyGroup.cs#L20>.
+
+The way the .NET SDK uses this is non breaking. It's never parsed since ProjectFileDependencyGroup already uses FrameworkName as a framework, so as long as this value and the targets name are changed together, it's good enough.
+
+1. `$/projectFileDependencyGroups`
+
+    ```diff
+      "projectFileDependencyGroups": {
+    -    "net8.0": [],
+    +    "production": [],
+      },
+    ```
+
+##### "project" PackageSpec changes
+
+The PackageSpec type and it's serialized form is used in multiple places in the NuGet code.
+It is a part of the assets file, but it is also used for the dgspec and the no-op scenarios.
+
+The PackageSpec is not versioned and at this point, I'm not proposing adding a version.
+
+The changes being made here can be divided into 2 parts.
+
+- Additive changes
+- Breaking changes
+
+The additive changes will be an always thing. Every newly serialized PackageSpec will make this change. This should not affect any of the existing scenarios, because at worse the new property is not parsed.
```

Comment:
```suggestion
The additive changes will be an always thing. Every newly serialized PackageSpec will make this change.
This should not affect any of the existing scenarios, because at worse the new property is not parsed.
```